### DISCUSSION TITLE
Please add Optional Separator in numbers frontend representation

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/frontend#7808] [Dashboard]
Since we now have formatted numbers, and an issue report on that https://github.com/home-assistant/frontend/issues/7745, leading to a lot of response which  was abruptly closed, please consider adding an option for the separator.

I'd really love to get the actual value in the frontend, instead of a formatted presentation of that. Fully aware many users might not care, or indeed want the separator. I feel there is a strong use case not to use the separator, and would hope something like the excel option for a separator (scheidingsteken) could be possible:

<img width="590" alt="Schermafbeelding 2020-11-25 om 11 09 09" src="https://user-images.githubusercontent.com/33354141/100214970-c968c180-2f10-11eb-8d22-c239e683915d.png">

when a backend value is 6551

<img width="386" alt="Schermafbeelding 2020-11-25 om 09 33 47" src="https://user-images.githubusercontent.com/33354141/100213906-84905b00-2f0f-11eb-94c0-a00b9729a123.png">

I would like the template

<img width="1092" alt="Schermafbeelding 2020-11-25 om 09 34 55" src="https://user-images.githubusercontent.com/33354141/100213960-94a83a80-2f0f-11eb-9b83-886ec1a133e1.png">

to show that in the frontend 

<img width="484" alt="Schermafbeelding 2020-11-25 om 11 25 47" src="https://user-images.githubusercontent.com/33354141/100215224-092fa900-2f11-11eb-82d7-f3d2ac4be50a.png">

and not 

<img width="476" alt="Schermafbeelding 2020-11-25 om 11 25 52" src="https://user-images.githubusercontent.com/33354141/100215260-12b91100-2f11-11eb-96ac-44440904e400.png">

my pardon for the non matching numbers, Sun just hit the Laundry...

**Comment by KTibow** [comment]
Do you think this should be per-user, per-device, or per-configuration?

**Reply by zsarnett** [reply]
If it was something we decided to add it would be a per user (like the language in the profile setting that would be set per device, as its set in the local storage)

**Reply by bramkragten** [reply]
(language is a user setting, it is sent to the backend)

**Comment by Mariusthvdb** [comment]
yes, I was thinking about that too Zack, per user in the profile settings.

I do think this (the frontend) should be in the hands of the end-user ultimately. Since per 118 the frontend manipulates the outcome of our carefully composed templates, more options for the enduser would be very much appreciated.

I know there are options planned I the backend templates for this too, but have no knowledge of how this would then be represented in the frontend.

Anyways, a final option in the profile would fix this 100%.
thanks for considering in the first place!

and just to show you why this is not a personal thing, but the common style used in my language setting:

![20201126_131907000_iOS](https://user-images.githubusercontent.com/33354141/100356640-ca265400-2ff3-11eb-8a3f-00be1711d6ef.jpg)

or in English btw:

<img width="1081" alt="Schermafbeelding 2020-11-26 om 14 28 58" src="https://user-images.githubusercontent.com/33354141/100356677-d6aaac80-2ff3-11eb-8427-b27777cc812f.png">

not a single separator in sight..

**Reply by KTibow** [reply]
If it's common for your language, couldn't you do this based on languages?

**Reply by Mariusthvdb** [reply]
think the point I'm making is, the separator might be common in computer lingo, but in many if not most other media isn't used by default. Hence an option to at least Not use it would be most obvious?

whether this can be done based on languages is not up to me, Zack and probably Bram would know best how to. I guess the language setting now changes it from 1.234,56 (Dutch) to 1,234.56 (English), so both using the separator. Again, to change that would need an option, for both.

**Comment by JeroenDeckers** [comment]
+1 here in Belgium same thing

**Reply by KTibow** [reply]
There's a button for that: 
![image](https://user-images.githubusercontent.com/10727862/100546911-c1fb2e00-3218-11eb-8441-42028d89129e.png)

**Comment by karwosts** [comment]
~~This seems long ago resolved.~~

**Reply by Mariusthvdb** [reply]
but how? there's is still no option to not show the separator in the profile settings

<img width="710" height="467" alt="Scherm­afbeelding 2025-10-21 om 09 40 40" src="https://github.com/user-attachments/assets/c380bfaf-ba5c-40c6-a03c-84892762d6df" />

we can set each individual entity to not show decimals, but that would require a great deal of work....

the other way around would be optimal (for my request):
in Profile set decimals to none
if needed, set a precision on an individual entity

**Reply by karwosts** [reply]
This isn't what you're looking for? 

<img width="565" height="250" alt="image" src="https://github.com/user-attachments/assets/6485abd4-d240-42be-ac37-cb4391a9e966" />

**Reply by Mariusthvdb** [reply]
at first glance that is what I believe too indeed, but as you can see it shows a decimal dot between 7 and 8.

even figured tot to be the thousand indicator or so, but all number entities still show a decimal.

Maybe I am completely missing the point, but as far as I can see we have no way in the number settings to disable that decimal globally 

Geen (none) should be 123456789 and not 1234567.89

**Reply by karwosts** [reply]
Ok I thought this was just about the thousands seperator... why would you remove the decimal seperator? Then you can't even differentiate fifteen versus one-and-a-half, it totally obfuscates the result.

I see now that None doesn't use your localized decimal seperator though, so yeah maybe that's not what you wanted.

**Reply by Mariusthvdb** [reply]
of all of the number entities I'd like to show decimals on a fraction of those

all of my energy/power sensors can be in whole numbers for what I care. as would brightness stuff .

Probably some volume levels that are in 0 to 1 scale need decimals, so Id set that on those individually

timing might need some thought, depending on the scale being seconds or minutes/hours 

temperature might be the most interesting class for decimals, but I set those to show but 1, as no sensor is actually that precise to begin with....

guess what this boils down to is when do we need to clutter the views with imprecise decimals

> Then you can't even differentiate fifteen versus one-and-a-half, it totally obfuscates the result.

I dont understand why you'd say that.  15 would show as 15, one-and-half would show as 2 probably, depending on your rules of rounding. the device class would make that either a great loss (in which case one would override the global setting to show 1 decimal), or completely sufficient (power/energy/brightness, even temperature...)

**Reply by karwosts** [reply]
I guess I got confused, at first you were talking about removing the thousands seperator with the laundry/lumens example, and then at some point this became about rounding decimals.

**Reply by Mariusthvdb** [reply]
Tbh, you’re probably right about the confusion….
I did show the correct decimal setting in the opening post but next continued with a separator….

separator is indeed settled.
Decimal not as far as I am concerned
